lib/oli_web/live/community_live/show.ex

```diff
@@ -23,37 +31,43 @@ defmodule OliWeb.CommunityLive.Show do
 
   def mount(%{"community_id" => community_id}, _session, socket) do
     socket =
-      case Groups.get_community(community_id) do
+      with %Community{status: status} = community <- Groups.get_community(community_id),
```

Comment:
Does not work with `%Community{status: "active"}`? Just asking

Reply:
+1

Reply:
If I do that it wouldn't work the way I want. Here I want to validate that the community exists and that it is `active` before rendering the page. If it doesn't meet any of these conditions, we wan't to render the index page with a flash message, but this message is different depending on the case. In order to match each case separately in the `else` clause, we need to have two separate checks.

An alternative could be to define a private method `validate_community(community_id)` that returns `{:ok, community}` if everything is fine, and `{:error, message}` if the community doesn't exist or is deleted. Wdyt?

Reply:
I think I would change `get_community` to return only an active community. And if they try to view a community that's not active - let the app 404 (same as if they tried an id that didn't exist) because they shouldn't be able to see a link to an inactive community to begin with right?